### PR TITLE
Add support for custom SQS consumer lock and disabling consumer

### PR DESCRIPTION
**Pull request by lcharkiewicz**
Adding support for custom SQS consumer lock name and disabling a consumer. 

Changelog: Adds support for custom SQS consumer lock name and disabling a consumer.

Goal: gravitational/cloud#10164

### CODE REVIEW

File: lib/events/athena/athena.go [hunk 1]
```diff
@@ -415,6 +426,17 @@ func (cfg *Config) SetFromURL(url *url.URL) error {
 		}
 		cfg.BatchMaxInterval = dur
 	}
+	consumerLockName := url.Query().Get("consumerLockName")
+	if consumerLockName != "" {
```
Comment (espadolini) [inline lib/events/athena/athena.go:430]:
```suggestion
	if consumerLockName := url.Query().Get("consumerLockName"); consumerLockName != "" {
```

File: lib/events/athena/athena.go [hunk 2]
```diff
@@ -415,6 +426,17 @@ func (cfg *Config) SetFromURL(url *url.URL) error {
...
+		cfg.ConsumerLockName = fmt.Sprintf("athena_%s", consumerLockName)
```
Comment (espadolini) [inline lib/events/athena/athena.go:431]:
```suggestion
		cfg.ConsumerLockName = "athena_" + consumerLockName
```

**Review by espadolini — APPROVED** [review]
I have no strong opinions on the names used, LGTM

File: lib/events/athena/athena.go (hunk 2 above)
Comment (espadolini) [reply lib/events/athena/athena.go:431]: Alternatively, given the potential for disallowed characters (such as `/`):
```suggestion
		cfg.ConsumerLockName = fmt.Sprintf("athena_%x", consumerLockName)
```

File: lib/events/athena/consumer.go [hunk 3]
```diff
@@ -251,7 +253,7 @@ func (c *consumer) runContinuouslyOnSingleAuth(ctx context.Context, eventsProces
 			err := backend.RunWhileLocked(ctx, backend.RunWhileLockedConfig{
 				LockConfiguration: backend.LockConfiguration{
 					Backend:  c.backend,
-					LockName: "athena_lock",
+					LockName: c.consumerLockName,
```
[the remark this answers is not in the document]
Comment (lcharkiewicz) [reply lib/events/athena/consumer.go]: Updated, thanks.

File: lib/events/athena/athena.go (hunk 2 above)
Comment (lcharkiewicz) [reply lib/events/athena/athena.go]: Ah, nice. Updated per suggestion, PTAL.

**Comment by lcharkiewicz** [comment]
Updated to use LockNameComponents and disabling a consumer. Let me know if that part requires more tests.

File: lib/events/athena/consumer.go [hunk 4]
```diff
@@ -258,7 +260,7 @@ func (c *consumer) runContinuouslyOnSingleAuth(ctx context.Context, eventsProces
 			err := backend.RunWhileLocked(ctx, backend.RunWhileLockedConfig{
 				LockConfiguration: backend.LockConfiguration{
 					Backend:            c.backend,
-					LockNameComponents: []string{"athena_lock"},
+					LockNameComponents: c.consumerLockName,
```
Comment (espadolini) [inline lib/events/athena/consumer.go:263]: Do we really need lock names like `athena/foo`? Even so, we can just do something like
```go
lockName := []string{"athena", c.consumerLockName}
if c.consumerLockName == "" {
  lockName = []string{"athena_lock"}
}
...
    LockNameComponents: lockName,
```

Alternatively, just `LockNameComponents: []string{"athena_" + cmp.Or(c.consumerLockName, "lock")}` would work, if we're fine with `athena_foo` as a lock name.

**Review by espadolini — APPROVED** [review]
I don't like how we pass along an arbitrary sequence of lock name components when we know it's either `athena_lock` or `athena/<foo>`, the rest all seems good.

**Comment by espadolini** [comment]
Remember to also update the athenaevents docs (in a separate PR so reviewer checks won't complain), depending on if (and when) we intend to support this for onprem.

**Comment by espadolini** [comment]
(we are not supporting `bypassSNS` just yet, if that makes the decision easier)

File: lib/events/athena/athena.go [hunk 5]
```diff
@@ -125,6 +125,12 @@ type Config struct {
 	BatchMaxItems int
 	// BatchMaxInterval defined interval at which parquet files will be created (optional).
 	BatchMaxInterval time.Duration
+	// ConsumerLockName defines a name of a SQS consumer lock (optional).
+	// If provided, it will be prefixed with "athena_" to avoid accidental
```
[the remark this answers is not in the document]
Comment (lcharkiewicz) [reply lib/events/athena/athena.go]: Thanks, updated.

File: lib/events/athena/athena.go [hunk 6]
```diff
@@ -125,6 +125,12 @@ type Config struct {
...
+	// collision with existing locks.
+	ConsumerLockName []string
```
[the remark this answers is not in the document]
Comment (lcharkiewicz) [reply lib/events/athena/athena.go]: Updated per suggestion.

File: lib/events/athena/consumer.go (hunk 4 above)
Comment (lcharkiewicz) [reply lib/events/athena/consumer.go]: > Do we really need lock names like `athena/foo`

I like the idea of using `LockNameComponents` for constructing locks and taking care of disallowed characters.

Updated per your suggestion, thanks!

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**